integreat_cms/cms/views/pois/poi_form_view.py

```diff
@@ -228,6 +234,28 @@ def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
                     ),
                 )
 
+            # Look explicitly for the primary contact, not any first one,
+            # as we do not delete non-primary contacts when deactivating contact in a region.
+            # "get()" is not used as it raises an exception if there is no primary contact.
+            contact = poi.contacts.filter(area_of_responsibility="").first()
```

Comment:
Maybe we could think about making this a manager method? This would then look something like:

```
class ContactQuerySet(model.QuerySet):
    def get_primary(self):
        return self.filter(area_of_responsibility="").first()

class Contact(AbstractBaseModel):
    ...
    objects = models.Manager.from_queryset(ContactQuerySet)()
```
Then we could do something like:
```
contact = poi.contacts.get_primary()
```
It's more code now, but maybe more extendable? :shrug: But we can definitely also keep it the way you did it, I don't think that this is too much duplicate code. Just a suggestion from the sideline :)

Reply:
Good idea 😸 Opened an issue for that #3922